src/mixins/autoLogout.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * @copyright Copyright (c) 2023 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import axios from '@nextcloud/axios'
+import { loadState } from '@nextcloud/initial-state'
+import { generateUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+
+const config = loadState('core', 'config', {})
+
+const getInterval = () => {
+	let interval = NaN
+	if (config.session_lifetime) {
+		interval = Math.floor(config.session_lifetime / 2)
+	}
+
+	return Math.min(
+		24 * 3600,
+		Math.max(
+			60,
+			isNaN(interval) ? 900 : interval
+		)
+	)
+}
+
+export default {
+
+	data() {
+		return {
+			autoLogoutInterval: null,
+			autoLogoutRegistered: false,
+		}
+	},
+
+	mounted() {
+		if (!getCurrentUser()) {
+			return
+		}
+
+		if (config?.auto_logout || !config?.session_keepalive) {
+			this.autoLogoutInterval = setInterval(this.registerCheck, 10000)
+		}
+	},
+
+	beforeDestroy() {
+		if (this.autoLogoutInterval) {
+			clearInterval(this.autoLogoutInterval)
+		}
+	},
+
+	methods: {
+		extendAutoLogout() {
+			const oldValue = localStorage.getItem('lastActive')
+			const newValue = Date.now().toString()
+			localStorage.setItem('lastActive', newValue)
+			const event = new StorageEvent('storage', {
+				key: 'lastActive',
+				oldValue,
+				newValue,
+			})
+			window.dispatchEvent(event)
+		},
+		registerCheck() {
+			if (!this.autoLogoutRegistered && this.postMessage) {
```

Comment:
I don't understand where `this.postMessage` and `this.postMessage.registerPostMessageHandler()` come from :thinking: 

Same for `this.sendPostMessage()` in the next if block.

Reply:
They are methods of the Office.vue component, this mixin is only used there and just to group related code together